spec/services/subscriptions/activate_service_spec.rb

```diff
@@ -7,45 +7,44 @@
 
   let(:timestamp) { Time.current }
 
-  describe 'activate_all_expired' do
-    let(:active_subscription) { create(:subscription) }
-    let(:pending_subscriptions) { create_list(:pending_subscription, 3, subscription_at: timestamp) }
-
-    let(:future_pending_subscriptions) do
+  describe '.activate_all_pending' do
+    it 'activates all pending subscriptions with subscription date set to today' do
+      create(:subscription)
+      create_list(:pending_subscription, 2, subscription_at: timestamp)
+      create(:pending_subscription, subscription_at: timestamp, plan: create(:plan, pay_in_advance: true))
       create_list(:pending_subscription, 2, subscription_at: (timestamp + 10.days))
-    end
 
-    before do
-      active_subscription
-      pending_subscriptions
-      future_pending_subscriptions
-    end
-
-    it 'activates all pending subscriptions with subscription date set to today' do
       expect { activate_service.activate_all_pending }
         .to change(Subscription.pending, :count).by(-3)
         .and change(Subscription.active, :count).by(3)
+        .and have_enqueued_job(SendWebhookJob).exactly(3).times
+        .and have_enqueued_job(BillSubscriptionJob).exactly(1).time
     end
 
-    it 'enqueues a SendWebhookJob' do
-      expect do
-        activate_service.activate_all_pending
-      end.to have_enqueued_job(SendWebhookJob).at_least(1).times
-    end
 
+    # Question: I'm not sure about this test, I'd expect another pending subscription to not be activated
     context 'with customer timezone' do
       let(:timestamp) { DateTime.parse('2023-08-24 00:07:00') }
-      let(:pending_subscription) { pending_subscriptions.first }
-
-      before do
-        pending_subscription.customer.update!(timezone: 'America/Bogota')
-        pending_subscription.update!(subscription_at: DateTime.parse('2023-08-24 04:17:00'))
+      let!(:pending_subscription) do
+        create(:pending_subscription, subscription_at: timestamp, customer: create(:customer, timezone: 'America/Bogota'))
```

Comment:
Layout/LineLength: Line is too long. [122/120] (https://rubystyle.guide#max-line-length)